vernac/g_vernac.mlg

```diff
@@ -442,22 +442,22 @@ GRAMMAR EXTEND Gram
   ;
   record_binder:
     [ [ id = name -> { (NoInstance,AssumExpr(id, [], CAst.make ~loc @@ CHole (None, IntroAnonymous, None))) }
-      | id = name; f = record_binder_body -> { f id } ] ]
+      | id = name; f = field_body -> { f id } ] ]
   ;
   assum_list:
-    [ [ bl = LIST1 assum_coe -> { bl } | b = simple_assum_coe -> { [b] } ] ]
+    [ [ bl = LIST1 assum_coe -> { bl } | b = assumpt -> { [b] } ] ]
   ;
   assum_coe:
-    [ [ "("; a = simple_assum_coe; ")" -> { a } ] ]
+    [ [ "("; a = assumpt; ")" -> { a } ] ]
   ;
-  simple_assum_coe:
-    [ [ idl = LIST1 ident_decl; oc = of_type_with_opt_coercion; c = lconstr ->
+  assumpt:
+    [ [ idl = LIST1 ident_decl; oc = of_type; c = lconstr ->
```

Comment:
By analogy with open/closed binders, what about respectively:
- assum_list -> assum_list (or even assumptions)
- assum_coe -> closed_assum_list (or closed_assumptions)
- assumpt -> open_assum_list (or open_assumptions)

Reply:
`assum_list` and `assum_coe` are spliced in the doc.

`assumpt` - perhaps.  I never learned exactly what this syntax means--the syntax update of the doc has not included explaining every bit of the syntax.  We can only do so much in each pass of revisions of the text, though I hope to work on adding and improving descriptions of the syntax soon.  It seemed to be defining a single assumption.

(Also, by the way, is it really useful to allow a `uinv_decl` here?  It seems a little funny, but I don't universes very well.)

![image](https://user-images.githubusercontent.com/1253341/96646360-e3dfc780-12e0-11eb-94d0-9f2093ed6879.png)

Reply:
I meant that there is a parallel with the syntax for binders where you can write `forall x y z : nat, ...` w/o parentheses (open-syntax binders, called `open_binders` in the refman) or you can write `forall (x y z : nat) (x' y' : bool), ...` with parentheses (closed-syntax binders, called `binder+` in the refman).

So, if we follow the example of [`open_binders`](https://coq.inria.fr/refman/language/core/assumptions.html#grammar-token-open-binders) in the doc, this could be something like:
```
open_assumptions ::= {+ ident_decl} of_type
                  |  {+ ( {+ ident_decl} of_type )}
```

About the `univ_decl`, I believe it needs to be supported (to indicate a universe polymorphic axiom, afaiu).

Note that `Context` is in some sense better designed here. Incidentally, I also see a possible improvement in the refman: `open_binders::=name+ : term` could be `open_binders::=name+ : type`.

Reply:
I'm not following you.  Are you proposing changing the names of existing nonterminals or extending the language?  I only want to do renaming in this PR--that's complicated enough.  Extensions, no matter how worthy, should be in separate PRs.

Reply:
I'm proposing a renaming: `assumpt` could be `open_assumptions`, and, in `g_vernac.mlg`, `assum_list` be `assumptions` and `assum_coe` be `closed_assumptions`.

This is to avoid having different abbreviations of `assumption` and to follow what is done for binders`.

Reply:
OK, making more sense.  This part still confuses me: when I look at the syntax it seems to me that `Axiom assumpt` defines a single assumption, in this case, a single Axiom.  Looking at all the .v files, I only ever see a single `ident_decl` present in `assumpt` (and only 2 cases with `{+ ( assumpt ) }`.  So I don't get why it should be `open_assumptions`, plural.

![image](https://user-images.githubusercontent.com/1253341/97059811-69b07c80-1546-11eb-88e2-9ea32372e345.png)

![image](https://user-images.githubusercontent.com/1253341/97059956-d1ff5e00-1546-11eb-8d64-bc0c651681f7.png)

Reply:
> Looking at all the .v files, I only ever see a single `ident_decl` present in `assumpt`

The syntax `Parameters (a b:nat) (c d:bool)` or `Variables (a b:nat) (c d:bool)` is indeed maybe rare in practice, knowing that the `Context` syntax, which is closer to the syntax of binders, is somehow more intuitive. Maybe could we unify both syntaxes (i.e. accept `Context a b : nat` on one side and accepts `Variables {a b : nat}` on the other side, or at least the later).

Reply:
Perhaps.  I've not spent much time thinking about how to make the grammar more consistent/unified, so I don't have an opinion on that.  Also a bit hard to evaluate without seeing the exact changes.

What shall we do with `assumpt`?  As I said:

> This part still confuses me: when I look at the syntax it seems to me that `Axiom assumpt` defines a single assumption, in this case, a single Axiom.  ...  So I don't get why it should be `open_assumptions`, plural.

I hope we can complete this PR soon.

Reply:
Ok, so keep it as it is. We can think about it later.